src/wallet/rpcwallet.cpp

```diff
@@ -2568,6 +2569,17 @@ static UniValue createwallet(const JSONRPCRequest& request)
     if (!request.params[2].isNull()) {
         empty = request.params[2].get_bool();
     }
+    SecureString passphrase;
+    passphrase.reserve(100);
+    if (!request.params[3].isNull()) {
+        passphrase = request.params[3].get_str().c_str();
+        if (!passphrase.empty()) {
+            if (empty) {
```

Comment:
These if statements could be combined to `if (!passphrase.empty() && empty)`

Reply:
I don't think they can. We want to set `empty` to `true` iff the passphrase is not the empty string. But we want to have an error if `empty` was already `true`. By combining these two, we could end up setting `empty` to `true` for a passphrase that is the empty string.